internal/plugin/connectors/tcp/mssql/connector_test.go

```diff
@@ -1,59 +1,108 @@
 package mssql
 
 import (
-	"context"
+	"errors"
+	"io"
 	"net"
 	"testing"
 
+	errorspkg "github.com/pkg/errors"
 	"github.com/stretchr/testify/assert"
 
 	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql/mock"
-	logmock "github.com/cyberark/secretless-broker/pkg/secretless/log/mock"
+	"github.com/cyberark/secretless-broker/internal/plugin/connectors/tcp/mssql/types"
 	pluginconnector "github.com/cyberark/secretless-broker/pkg/secretless/plugin/connector"
 	"github.com/denisenkom/go-mssqldb"
 )
 
-func TestHappyPath(t *testing.T) {
-	logger := logmock.NewLogger()
+func TestSingleUseConnector_Connect(t *testing.T) {
+	t.Run("singleUseConnector.driver#Connect success", func(t *testing.T) {
+		expectedBackendConn := mock.NewNetConn(nil)
+
+		connector := NewSingleUseConnectorWithOptions(
+			mock.DefaultConnectorOptions(),
+			mock.MSSQLConnectorCtor(
+				func(options *mock.MSSQLConnectorCtorOptions) {
+					options.BackendConn = expectedBackendConn
+				},
+			),
+		)
+		actualBackendConn, err := runDefaultTestConnect(connector)
+
+		assert.Nil(t, err)
+		if err != nil {
+			return
+		}
+
+		assert.Equal(t, expectedBackendConn, actualBackendConn)
+	})
+
+	t.Run("singleUseConnector.driver#Connect fail", func(t *testing.T) {
+		methodFails(t, mock.MSSQLConnectorCtor(
+			func(ctorOptions *mock.MSSQLConnectorCtorOptions) {
+				ctorOptions.Err = methodFailsExpectedErr
+			}),
+		)
+	})
+}
+// Test helpers
+//
+
+// runDefaultTestConnect passes in default values to the Connect method of a connector.
+// This helps avoid boilerplate
+func runDefaultTestConnect(
+	connector *SingleUseConnector,
+) (net.Conn, error) {
 	clientConn := mock.NewNetConn(nil)
-	expectedBackendConn := mock.NewNetConn(nil)
 	creds := pluginconnector.CredentialValuesByID{
 		"credName": []byte("secret"),
 	}
 
-	ctor := mock.NewSuccessfulMSSQLConnectorCtor(
-		func(ctx context.Context) (net.Conn, error) {
-			interceptor := mssql.ConnectInterceptorFromContext(ctx)
-
-			interceptor.ServerPreLoginResponse <- map[uint8][]byte{ 0: {0, 0} }
+	return connector.Connect(clientConn, creds)
+}
 
-			<- interceptor.ClientLoginRequest
+// methodFailsExpectedErr is the error value used inside methodFails
+var methodFailsExpectedErr = errors.New("failed for the test")
 
-			interceptor.ServerLoginResponse <- &mssql.LoginResponse{}
+// methodFails checks that the expected error is present in:
+// 1. the error returned from the call to the Connect method
+// 2. the error propagated to the client
+func methodFails(
+	t *testing.T,
+	connectorOption types.ConnectorOption,
```

Comment:
Quote @diverdane 
> Just a thought... but could we use a variadic connectorOptions here instead of a single connectorOption? Not sure if this would ever come into play, but if it's cheap and easy to change